cli/flox-rust-sdk/src/providers/flox_cpp_utils.rs

```diff
@@ -333,14 +386,61 @@ mod tests {
             env.install(&pkgs, &flox)
         });
         if let Err(e) = res {
+            eprintln!("Error: {:?}", e);
             let err_string = e.to_string();
-            let has_nix_error = err_string.contains("is marked as broken")
+            let has_nix_error = err_string.contains("I'm broken inside")
                 || err_string.contains("cached failure of attribute");
             assert!(has_nix_error);
         } else {
             panic!("expected an error");
         }
     }
 
+    #[test]
+    fn fills_in_priority() {
+        let locked_hello = r#"
+        {
+            "broken": false,
+            "derivation": "/nix/store/4w0wsrlfad3ilqjxk34fnkmdckiq0k0m-hello-2.12.1.drv",
+            "description": "Program that produces a familiar, friendly greeting",
+            "flake-description": "A collection of packages for the Nix package manager",
+            "licenses": [
+                "GPL-3.0-or-later"
+            ],
+            "locked-flake-attr-path": "legacyPackages.aarch64-darwin.hello",
+            "locked-url": "github:NixOS/nixpkgs/56bf14fe1c5ba088fff3f337bc0cdf28c8227f81",
+            "name": "hello-2.12.1",
+            "output-names": [
+                "out"
+            ],
+            "outputs": {
+                "out": "/nix/store/ia1pdwpvhswwnbamqkzbz69ja02bjfqx-hello-2.12.1"
+            },
+            "outputs-to-install": [
+                "out"
+            ],
+            "package-system": "aarch64-darwin",
+            "pname": "hello",
+            "priority": null,
+            "requested-outputs-to-install": null,
+            "system": "aarch64-darwin",
+            "unfree": false,
+            "version": "2.12.1"
+        }
+        "#;
+        let locked: LockedInstallable = serde_json::from_str(locked_hello).unwrap();
+        let descriptor = ManifestPackageDescriptorFlake {
+            flake: "github:NixOS/nipxkgs#hello".to_string(),
+            priority: Some(10),
+            systems: None,
+        };
+        let mock_locker = InstallableLockerMock::new();
+        mock_locker.push_lock_result(Ok(locked));
+        let resolved = mock_locker
```

Comment:
suggestion blocking: isn't this testing the mock locker, not the actual locker? Seems like we should have a unit test for `set_priority`. Or we need to push the mocking down to the JSON layer so that `lock_flake_installable` gets back mock JSON

Reply:
It's using the mock locker to test the `set_priority` function, which is used in both lockers. A unit test for `set_priority` would also be fine with me. I don't think adding that extra layer of mocking is in scope for this PR.

Reply:
I think testing `set_priority` is less brittle

Reply:
Yeah I'm doing it now